npmignore should trump files array

zkat opened the pull request:
currently, anything explicitly in files array gets included, regardless of `.npmignore` lines

**WIP** -- this is only a breaking test so far.

This depends on #11292

iarna committed:
tacks@1.0.9

Tool to generate fixtures from folders and module for creating and tearing
down filesystem fixtures easily.

Credit: @iarna
PR-URL: https://github.com/npm/npm/pull/11292
Reviewed-By: @zkat
Reviewed-By: @othiym23

iarna committed:
test: Port legacy tests to tap

This removes all the relatively cryptic legacy tests and creates
new tap tests that check the same functionality.

Credit: @iarna
Credit: @zkat
PR-URL: https://github.com/npm/npm/pull/11292
Reviewed-By: @iarna
Reviewed-By: @zkat
Reviewed-By: @othiym23

iarna committed:
test: We don't leak auth info into the env

Credit: @zkat
PR-URL: https://github.com/npm/npm/pull/11292
Reviewed-By: @iarna

zkat committed:
test: check that env vars properly passed to scripts

* removed legacy-env-reader test
* test that regular env vars are passed down
* test that npm-prefixed vars are stripped
* test that package.json configs are added to env
* test that package.json configs are prefixed with npm_package
  and added to the env

Credit: @zkat
PR-URL: https://github.com/npm/npm/pull/11292
Reviewed-By: @iarna

iarna committed:
test: add npm self installation test

Credit: @iarna
PR-URL: https://github.com/npm/npm/pull/11292
Reviewed-By: @zkat

wyze committed:
doc: Add colon for consistency

PR-URL: https://github.com/npm/npm/pull/11414
Credit: @wyze
Reviewed-By: @zkat

wyze committed:
doc: Remove confusing ref of adduser

Credit: @wyze
PR-URL: https://github.com/npm/npm/pull/11416
Reviewed-By: @zkat

zkat committed:
test: tests for file inclusion and .npmignore

Credit: @zkat
PR-URL: https://github.com/npm/npm/pull/11292
Reviewed-By: @iarna

zkat committed:
test: npmignore should trump files array

Commit:
ls: Only list runtime dependencies when doing `npm ls --production`

Fixes: #11559
Credit: @yibn2008
Reviewed-By: @iarna
PR-URL: https://github.com/npm/npm/pull/11562

iarna commented:
wow this one is gonna kinda suck to get right from a UX perspective =/

'cause yeah, if I have folders in my `files` section then I def want `.npmignore` to help me out. But if I have explicit files listed in my `files` section and they get skipped due to a `.npmignore` I'm gonna be _really_ surprised.

files: ```
docs/README.txt
```
npmignore:
```
*.txt
```

Ok that example is extremely contrived. My gut instinct here is that this is gonna burn folks, but I'm going to have to think more about what a real world example would look like.

rhendric committed:
update: don't attempt to update linked packages

Fixes: #11362
PR-URL: https://github.com/npm/npm/pull/11584
Credit: @rhendric
Reviewed-By: @iarna

iarna committed:
node-gyp@3.3.0

Update to a more recent version that uses a version of npmlog compatible
with npm itself.

Credit: @rvagg
Reviewed-By: @zkat
PR-URL: https://github.com/npm/npm/pull/11528

iarna committed:
dep: npmlog subsubdep

gauge@1.2.7: Gauge update to newer lodash versions

Credit: @iarna

iarna committed:
tap@5.6.0

iarna committed:
glob@7.0.0

Raise error if `options.cwd` is specified, and not a directory

Credit: @isaacs

iarna committed:
rimraf@2.5.2

Use glob@7.0.0

Credit: @isaacs

iarna committed:
lodash._baseuniq@4.4.0

iarna committed:
lodash.clonedeep@4.3.0

iarna committed:
lodash.isarguments@3.0.7

iarna committed:
lodash.keys@4.0.3

iarna committed:
lodash.union@4.2.0

iarna committed:
lodash.uniq@4.2.0

iarna committed:
lodash.without@4.1.0

iarna committed:
doc: update changelog for 3.7.4

iarna committed:
gen-changelog: Support multiple credits

iarna committed:
mailmap: add new folks

iarna committed:
3.7.4

iarna committed:
update AUTHORS

iarna committed:
deps: fix missing glob subdeps

When updating the top level npm to glob@7, the subdeps that
still depended on glob@6 got new versions installed but they
weren't added to the commit. This adds them back in.

Fixes: #11633
Credit: @iarna
Reviewed-By: @othiym23
PR-URL: https://github.com/npm/npm/pull/11638

iarna committed:
doc: update changelog for 3.7.5

iarna committed:
build: Make sure ls on a clean repo works w/o error

iarna committed:
3.7.5

zkat commented:
Closing again and remaking against master.
